src/shared/filesystemUtilities.ts

```diff
@@ -73,7 +73,11 @@ export async function findFileInParentPaths(searchFolder: string, fileToFind: st
 
 const mkdtemp = promisify(fs.mkdtemp)
 export const makeTemporaryToolkitFolder = async (...relativePathParts: string[]) => {
-    const _relativePathParts = relativePathParts || ['vsctk']
+    const _relativePathParts = relativePathParts || []
+    if (_relativePathParts.length === 0) {
+        _relativePathParts.push('vsctk')
+    }
+
     const tmpPath = path.join(tempDirPath, ..._relativePathParts)
```

Comment:
where does `tempDirPath` come from - if this is system wide you might run into a problem that the JetBrains toolkit faced : https://github.com/aws/aws-toolkit-jetbrains/issues/950

sam has a limitation that the 'template' and source folders must be relative within the same drive-mount. On windows if you create a temporary template on `C:\` for example but the source workspace is on `D:\` then it doesn't work.

You should update the temporary folder to use something relative to the workspace. I'm not sure if VS Code exposes this ability.

Reply:
https://github.com/aws/aws-toolkit-vscode/blob/9a9082736f3f665711083a793eaa8b2454216ba0/src/shared/filesystemUtilities.ts#L16

It's only used for tests and uses the Node `os.tmpdir` value.

Reply:
Let me verify this. We've never tried running with a second drive. I'll update the comments here and add an issue if there is one.

Reply:
@kiiadi  tested by making a SAM App on a D drive (windows and temp are on C drive), init, run (performs a build), debug (performs a build) were successful.